src/ast/irbuilderbpf.cpp

```diff
@@ -73,21 +73,35 @@ libbpf::bpf_func_id IRBuilderBPF::selectProbeReadHelper(AddrSpace as, bool str)
   return fn;
 }
 
+Value *IRBuilderBPF::CreateGetPid(const location &loc)
+{
+  Value *pidtgid = CreateGetPidTgid(loc);
+  Value *pid = CreateTrunc(CreateLShr(pidtgid, 32), getInt32Ty(), "pid");
+  return pid;
+}
+
+Value *IRBuilderBPF::CreateGetTid(const location &loc)
+{
+  Value *pidtgid = CreateGetPidTgid(loc);
+  Value *tid = CreateTrunc(pidtgid, getInt32Ty(), "tid");
+  return tid;
+}
+
 AllocaInst *IRBuilderBPF::CreateUSym(llvm::Value *val,
                                      int probe_id,
                                      const location &loc)
 {
   std::vector<llvm::Type *> elements = {
     getInt64Ty(), // addr
-    getInt64Ty(), // pid
-    getInt64Ty(), // probe id
+    getInt32Ty(), // pid
+    getInt32Ty(), // probe id
```

Comment:
Drive by fix?

Reply:
This is the core of the change - pid and probe id from 64 to 32 bits

Reply:
Yeah, I don't know what was going through my head when I asked that question.